engines/order_management/spec/services/order_management/stock/packer_spec.rb

```diff
@@ -10,7 +10,7 @@ module Stock
 
       subject { Packer.new(stock_location, order) }
 
-      before { order.line_items.first.variant.update(weight: 1) }
+      before { order.line_items.first.variant.weight = 1 }
```

Comment:
Why did we need to change this?

Reply:
I've just had a look at this again and done some debugging. Setting the variant's weight in this way is actually not how variant weight is set, it's a computed value. I tried adding some simpler cases where a variant's weight is updated this way, and they failed. I tried the same tests in master and they also failed. Eg:

```
variant.weight.update(weight: 1)
expect(variant.reload.weight).to eq 1
```

The test fails and result is zero in both this branch and master.

Updating a variant's weight correctly is done via `variant.update(unit_value: 1)`. I'll add a commit now.

Reply:
Ahhh nice sleuthing! 🎉